### PR TITLE
upgrade cachified

### PR DESCRIPTION
**Pull request by kentcdodds**
I've bumped into a little type issue. @Xiphe do you think you can help me figure out what's going on here?

### CODE REVIEW

File: app/utils/cache.server.ts [hunk 1]
```diff
@@ -155,13 +169,14 @@ export async function searchCacheKeys(search: string, limit: number) {
 
 export async function cachified<Value>({
 	timings,
-	reporter = verboseReporter(),
+	reporter = verboseReporter<Value>(),
```
Comment (kentcdodds) [inline app/utils/cache.server.ts:172]: The type error I'm getting on this line is:

> Type 'CreateReporter<Value>' is not assignable to type 'never'. ts(2322)

Comment (Xiphe) [reply app/utils/cache.server.ts:172]: I left a `reporter?: never` on `CachifiedOptions` in order to warn people that their reporter will not function here. 

To keep the old signature you'd likely want to do this:

```ts
export async function cachified<Value>({
	timings,
	reporter = verboseReporter<Value>(),
	...options
}: Omit<CachifiedOptions<Value>, 'reporter'> & {
	timings?: Timings
	reporter?: Reporter<Value>
}): Promise<Value> {
	return baseCachified(
		options,
		mergeReporters(cachifiedTimingReporter(timings), reporter),
	)
}
```

Or adopt the new signature with:

```ts
export async function cachified<Value>({
	timings,
	...options
}: CachifiedOptions<Value> & {
	timings?: Timings
}, reporter: Reporter<Value> = verboseReporter<Value>()): Promise<Value> {
	return baseCachified(
		options,
		mergeReporters(cachifiedTimingReporter(timings), reporter),
	)
}
```

Comment (kentcdodds) [reply app/utils/cache.server.ts:172]: I tried those, and with both of them I had the same error:

```
Type 'CreateReporter<Value>' is not assignable to type 'Reporter<Value>'.
  Types of parameters 'context' and 'event' are incompatible.
    Type 'CacheEvent<Value>' is not assignable to type 'Omit<Context<Value>, "report">'.
      Type 'GetFreshValueStartEvent' is missing the following properties from type 'Omit<Context<Value>, "report">': metadata, ttl, key, cache, and 6 more.ts(2322)
```

Comment (Xiphe) [reply app/utils/cache.server.ts:172]: urgs. I'm gonna have another look

Comment (Xiphe) [reply app/utils/cache.server.ts:172]: => https://github.com/epicweb-dev/epic-stack/pull/618

**Pull request closed without merging**